src/main/scala/com/microsoft/hyperspace/actions/RefreshActionBase.scala

```diff
@@ -120,7 +118,21 @@ private[actions] abstract class RefreshActionBase(
             _) =>
           location
             .allFiles()
-            .map(f => FileInfo(f.getPath.toString, f.getLen, f.getModificationTime))
+            .map { f =>
+              // For each file, if it already has a file id, add that id
+              // to its corresponding FileInfo. Note that if the content
+              // of an existing file is changed, it still keeps its id.
+              // This wont cause an issue as the file will show as
+              // deleted/appended and its index entries will be
+              // fixed during refresh.
+              val filePath = f.getPath.toString
+              previousIndexLogEntry.fileNameToIdMap.get(filePath) match {
+                case Some(id) =>
+                  FileInfo(filePath, f.getLen, f.getModificationTime, id)
+                case _ =>
+                  FileInfo(filePath, f.getLen, f.getModificationTime)
```

Comment:
How about assigning a new id here? for appended files?

Reply:
Please refer to this [answer](https://github.com/microsoft/hyperspace/pull/234#discussion_r517049427).
Lets keep the logic of generating/assigning-ids in one place.